test/framework/src/main/java/org/elasticsearch/test/InternalAggregationTestCase.java

```diff
@@ -312,6 +312,13 @@ public void testReduceRandom() {
             //check that non final reduction never adds buckets
             assertThat(reducedBucketCount, lessThanOrEqualTo(initialBucketCount));
             toReduce = new ArrayList<>(toReduce.subList(r, toReduceSize));
+            /*
+             * sometimes simulate cross cluster search by serializing and
+             * deserializing the partially reduced result.
+             */
```

Comment:
Should it simply say that we test the serialization of partially reduced result since you have another change that will make this possible in normal search ?

Reply:
Sure.